Mathlib/Analysis/SpecialFunctions/OrdinaryHypergeometric.lean

```diff
@@ -0,0 +1,221 @@
+/-
+Copyright (c) 2024 Edward Watine. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Edward Watine
+-/
+import Mathlib.Analysis.Analytic.OfScalars
+import Mathlib.Analysis.SpecificLimits.RCLike
+
+/-!
+# Ordinary hypergeometric function in a Banach algebra
+
+In this file, we define `ordinaryHypergeometric`, the _ordinary_ or _Gaussian_ hypergeometric
+function in a topological algebra `𝔸` over a field `𝕂` given by: $$
+_2\mathrm{F}_1(a\ b\ c : \mathbb{K}, x : \mathbb{A}) = \sum_{n=0}^{\infty}\frac{(a)_n(b)_n}{(c)_n}
+\frac{x^n}{n!}   \,,
+$$
+with $(a)_n$ is the ascending Pochhammer symbol (see `ascPochhammer`).
+
+This file contains the basic definitions over a general field `𝕂` and notation for `₂F₁`,
+as well as showing that terms of the series are zero if any of the `(a b c : 𝕂)` are sufficiently
+large non-positive integers, rendering the series finite. In this file "sufficiently large" means
+that `-n < a` for the `n`-th term, and similarly for `b` and `c`.
+
+- `ordinaryHypergeometricSeries` is the `FormalMultilinearSeries` given above for some `(a b c : 𝕂)`
+- `ordinaryHypergeometric` is the sum of the series for some `(x : 𝔸)`
+- `ordinaryHypergeometricSeries_eq_zero_of_nonpos_int` shows that the `n`-th term of the series is
+zero if any of the parameters are sufficiently large non-positive integers
+
+## `[RCLike 𝕂]`
+
+If we have `[RCLike 𝕂]`, then we show that the latter result is an iff, and hence prove that the
+radius of convergence of the series is unity if the series is infinite, or `⊤` otherwise.
+
+- `ordinaryHypergeometricSeries_eq_zero_iff` is iff variant of
+`ordinaryHypergeometricSeries_eq_zero_of_nonpos_int`
+- `ordinaryHypergeometricSeries_radius_eq_one` proves that the radius of convergence of the
+`ordinaryHypergeometricSeries` is unity under non-trivial parameters
+
+## Notation
+
+`₂F₁` is notation for `ordinaryHypergeometric`.
+
+## References
+
+See <https://en.wikipedia.org/wiki/Hypergeometric_function>.
+
+## Tags
+
+hypergeometric, gaussian, ordinary
+-/
+
+open Nat FormalMultilinearSeries
+
+section Field
+
+variable {𝕂 : Type*} (𝔸 : Type*) [Field 𝕂] [Ring 𝔸] [Algebra 𝕂 𝔸] [TopologicalSpace 𝔸]
+  [TopologicalRing 𝔸]
+
+/-- The coefficients in the ordinary hypergeometric sum. -/
+noncomputable abbrev ordinaryHypergeometricCoefficient (a b c : 𝕂) := fun n => ((n !⁻¹ : 𝕂) *
+    (ascPochhammer 𝕂 n).eval a * (ascPochhammer 𝕂 n).eval b * ((ascPochhammer 𝕂 n).eval c)⁻¹)
```

Comment:
We should follow the [Left of Colon](https://leanprover-community.github.io/contribute/style.html#hypotheses-left-of-colon) style here:
```suggestion
noncomputable abbrev ordinaryHypergeometricCoefficient (a b c : 𝕂) (n : ℕ) :=
  ((n !⁻¹ : 𝕂) * (ascPochhammer 𝕂 n).eval a *
    (ascPochhammer 𝕂 n).eval b * ((ascPochhammer 𝕂 n).eval c)⁻¹)
```